MapboxNavigation/NavigationViewController.swift

```diff
@@ -46,7 +43,15 @@ public protocol NavigationViewControllerDelegate: VisualInstructionDelegate {
      */
     @objc(navigationViewController:didArriveAtWaypoint:)
     optional func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt waypoint: Waypoint) -> Bool
-
+    
+    /**
+     Called when the user completes the final leg on the route.
+     
+     You can use this to detect the status of `CarPlayManager.isConnected`.
+     
+     */
+    @objc optional func navigationViewControllerDidFinishRoute(_ navigationViewController: NavigationViewController, byArrivingAtFinalStep isFinalLeg: Bool) -> Bool
```

Comment:
Can we try to make this self-explanatory? We already have didArrive so adding didFinishRoute is a bit confusing.
How about `…shouldDismissFeedback…` that defaults to true if unimplemented?